From 2ab4190f6a5629732a2f4e8efb6b61ce09402f58 Mon Sep 17 00:00:00 2001
From: Jackie264 <jackie.han@gmail.com>
Date: Mon, 14 Sep 2026 20:30:00 +0800
Subject: [PATCH] Update device tree for ipq9570 with memory and regulator
 changes

Added reserved memory configurations and adjusted regulator settings for improved performance. Disabled unused QCE nodes to prevent clock controller issues.

Signed-off-by: Jackie264 <jackie.han@gmail.com>
---
 .../linux/qualcommbe/dts/ipq9570-sbe1v1k.dts  | 41 +++++++++++++++++--
 1 file changed, 37 insertions(+), 4 deletions(-)

diff --git a/target/linux/qualcommbe/dts/ipq9570-sbe1v1k.dts b/target/linux/qualcommbe/dts/ipq9570-sbe1v1k.dts
index f39badf135a028..1a8be3632941d4 100644
--- a/target/linux/qualcommbe/dts/ipq9570-sbe1v1k.dts
+++ b/target/linux/qualcommbe/dts/ipq9570-sbe1v1k.dts
@@ -23,13 +23,25 @@
 
 	chosen {
 		stdout-path = "serial0:115200n8";
+		bootargs = "cma=256M";
 	};
 
 	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
 		tzapp:tzapp@49B00000 {
 			no-map;
 			reg = <0x0 0x49B00000 0x0 0x00600000>;
 		};
+
+		linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0x0 0x10000000>; /* 256MB = 0x10000000 */
+			linux,cma-default;
+		};
 	};
 
 	regulator_fixed_3p3: s3300 {
@@ -171,6 +183,19 @@
 	status = "okay";
 };
 
+/*
+ * No upstream driver currently binds these QCE nodes. Leaving them enabled
+ * prevents the GCC clock controller from completing sync_state(), which in
+ * turn leaves cpufreq-dt stuck in deferred probe.
+ */
+&crypto {
+	status = "disabled";
+};
+
+&cryptobam {
+	status = "disabled";
+};
+
 &rpm_requests {
 	regulators {
 		compatible = "qcom,rpm-mp5496-regulators";
@@ -184,13 +209,15 @@
 		* the regulators are brought up with 725mV which is sufficient for all the
 		* corner parts to operate at 800MHz
 		*/
-			regulator-min-microvolt = <725000>;
+			regulator-min-microvolt = <587500>;
 			regulator-max-microvolt = <1075000>;
+			regulator-always-on;
+			regulator-boot-on;
 		};
 
-		mp5496_l5: l5 {
-			regulator-min-microvolt = <1800000>;
-			regulator-max-microvolt = <1800000>;
+		ipq9574_s2: s2 {
+			regulator-min-microvolt = <700000>;
+			regulator-max-microvolt = <1000000>;
 			regulator-always-on;
 			regulator-boot-on;
 		};
@@ -497,6 +524,12 @@
 		reg = <16>;
 
 		qcom,package-mode = "qsgmii";
+		/*
+		 * Match the 300 mV PQSGMII TX amplitude programmed by the
+		 * stock SSDK. The generic 600 mV default causes receive CRC
+		 * errors on the QSGMII channel connected to PPE port 3.
+		 */
+		qcom,tx-drive-strength-milliwatt = <300>;
 
 		qca8075_18: ethernet-phy@18 {
 			compatible = "ethernet-phy-ieee802.3-c22";
